src/content/docs/en/reference/loader-reference.mdx

```diff
@@ -3,6 +3,324 @@ title: Astro Loader API
+
+The recommended pattern is to define a function that accepts configuration options and returns the loader object, in the same way that you would normally define an Astro integration or Vite plugin. 
+
+
+```ts title=loader.ts
+
+import type { Loader, LoaderContext } from 'astro/loaders';
+
+// Define any options that the loader needs
+export function myLoader(options: { url: string }): Loader {
+  // Configure the loader 
+  const feedUrl = new URL(options.url);
+  // Return a loader object
+  return {
+    name: "my-loader";
+    // Called when updating the collection.
+    load: async (context: LoaderContext): Promise<void> => {
+      // Load data and update the store
+    }
+    // Optionally, define the schema of an entry. 
+    // It will be overridden by user-defined schema. 
+    schema: async () => z.object({
+      // ...
+    })
+  };
+}
+```
+
+## Loader API
+
+The API for [inline loaders](#inline-loaders) is very simple, and is shown above. This section shows the API for defining an object loader.
+
+### The `Loader` object
+
+The loader object has the following properties:
+
+#### `name`
+**Type**: `string`
+
+A unique name for the loader, used in logs and [for conditional loading](#integration-support).
+
+#### `load`
+**Type**: `(context: LoaderContext) => Promise<void>`
+
+An async function that is called at build time to load data and update the store. See [`LoaderContext`](#loadercontext) for more information.
+
+#### `schema`
+**Type**: `ZodSchema | Promise<ZodSchema> | (() => ZodSchema | Promise<ZodSchema>)`
+
+An optional [Zod schema](/en/guides/defining-collections/#defining-datatypes-with-zod) that defines the shape of the entries. It is used to both validate the data, and also to generate TypeScript types for the collection. If a function is provided, it will be called at build time before `load()` to generate the schema. You can use this to dynamically generate the schema based on the configuration options or by introspecting an API. 
+
+### `LoaderContext`
+This object is passed to the `load()` method of the loader, and contains the following properties:
+
+#### `collection`
+**Type**: `string`
+
+The unique name of the collection. This is the key in the `collections` object in the `src/content/config.ts` file.
+
+#### `store`
+**Type**: [`DataStore`](#datastore)
+
+A database to store the actual data. Use this to update the store with new entries. See [`DataStore`](#datastore) for more information.
+
+#### `meta`
+**Type**: `MetaStore`
+
+A simple KV store, designed for things like sync tokens and last-modified times. Use this to store metadata about the collection. The data is persisted between builds, but is only available inside the loader.
```

Comment:
> The data is persisted between builds

It might be worth explaining where/how it is persisted. If my last modified time is persisted but the actual data that was read isn't, then it becomes a problem since the implementation won't fetch everything again.

For example, what should I configure in Turbo to store this information and the loaded entries?

Reply:
Good point. It is actually persisted in the data store, but a different scope so it's not available in the collection. Would something like "This metadata is persisted between builds alongside the collection data, but is only available inside the loader" work?